Vic2ToHoI4/Source/HOI4World/Names/Names.cpp

```diff
@@ -112,13 +113,21 @@ void HoI4::Names::Factory::addNamesToMap(std::map<std::string, std::vector<std::
 
 void HoI4::Names::Factory::checkForNames()
 {
-	for (const auto& maleNamesMapping: maleNames)
+	for (const auto& [culture, maleNamesInCulture]: maleNames)
 	{
-		auto culture = maleNamesMapping.first;
-
-		if (auto names = femaleNames.find(culture); (names == femaleNames.end()) || names->second.empty())
+		if (maleNamesInCulture.size() < 8)
+		{
+			Log(LogLevel::Warning) << "Too few male names for " << culture;
+		}
+		if (auto names = surnames.find(culture);
+			 (names == surnames.end()) || names->second.empty() || (names->second.size() < 3))
```

Comment:
names->second.empty() is redundant, since names->second.size() is 0 in that case.